vignettes/BiocParallel_BatchtoolsParam.Rmd

```diff
@@ -0,0 +1,229 @@
+SLURM template as an example:
+
+```{r}
+fname <- batchtoolsTemplate("slurm")
+cat(readLines(fname), sep="\n")
+```
+
+The `<%= =>` blocks are automatically replaced by the values of the elements in
+the `resources` argument in the`BatchtoolsParam` constructor. Failing to specify critical parameters
+properly (e.g., wall time or memory limits too low) will cause jobs to
+crash, usually rather cryptically. We suggest setting parameters
+explicitly to provide robustness to changes to system defaults. Note
+that `<%= =>` the blocks themselves do not usually need to be modified in the
+template.
+
+The part of the template that is most likely to require explicit
+customization is the last line containing the call to `Rscript`. A more
+customized call may be necessary if the R installation is not standard,
+e.g., if multiple versions of R have been installed on a cluster. For
+example, one might use instead:
+
+````{verbatim}
+    echo 'batchtools::doJobCollection("<%= uri %>")' |\
+        ArbitraryRcommand --no-save --no-echo
+````
+
+If such customization is necessary, we suggest making a local copy of
+the `BiocParallelParam` template, modifying it as required, and then constructing a object
+with the modified template using `template` the argument. However, we find that the
+default templates accessible with `batchtoolsTemplate` are satisfactory in most cases.
+
+# Use cases
+
+As an example for a BatchtoolParam job being run on an SGE cluster, we
+use the same `piApprox` function as defined earlier. The example runs the function
+on 5 workers and submits 100 jobs to the SGE cluster.
+
+```{r simple_sge_example, eval=FALSE}
+library(BiocParallel)
+
+## Pi approximation
+piApprox <- function(n) {
+    nums <- matrix(runif(2 * n), ncol = 2)
+    d <- sqrt(nums[, 1]^2 + nums[, 2]^2)
+    4 * mean(d <= 1)
+}
+
+template <- system.file(
+    package = "BiocParallel",
+    "unitTests", "test_script", "test-sge-template.tmpl"
+)
+param <- BatchtoolsParam(workers=5, cluster="sge", template=template)
+
+## Run parallel job
+result <- bplapply(rep(10e5, 100), piApprox, BPPARAM=param)
+```
+
+Example of SGE with minimal code:
+
+```{r demo_sge, eval=FALSE}
+<<demo_sge, eval=FALSE>>=
+library(BiocParallel)
+
+## Pi approximation
+piApprox <- function(n) {
+    nums <- matrix(runif(2 * n), ncol = 2)
+    d <- sqrt(nums[, 1]^2 + nums[, 2]^2)
+    4 * mean(d <= 1)
+}
+
+template <- system.file(
+    package = "BiocParallel",
+    "unitTests", "test_script", "test-sge-template.tmpl"
+)
+param <- BatchtoolsParam(workers=5, cluster="sge", template=template)
+
+## start param
+bpstart(param)
+
+## Display param
+param
+
+## To show the registered backend
+bpbackend(param)
+
+## Register the param
+register(param)
+
+## Check the registered param
+registered()
+
+## Run parallel job
+result <- bplapply(rep(10e5, 100), piApprox)
+
+bpstop(param)
+```
+
+# sessionInfo\(\)
+
+```{r sessionInfo, results="asis"}
+sessionInfo()
+```
```

Comment:
Here is an interesting one. The original writer added the syntax output as a curated list instead of simply showing the output. 2 edits. Please hide the output and add the list shown on the original vignette. I wrote the code up to make sure that it matched the OG. You can just copy it over: 
<img width="864" alt="Screen Shot 2022-09-14 at 12 57 10 PM" src="https://user-images.githubusercontent.com/105299892/190216391-c3677bb5-1766-4d93-b7fc-c78848a24836.png">

Reply:
When I did this and knit the file, it looks like this: 
<img width="776" alt="Screen Shot 2022-09-14 at 12 57 46 PM" src="https://user-images.githubusercontent.com/105299892/190216519-e6b2a2f8-788b-4875-8502-0c4be37afb81.png">

Reply:
I copied over the output from the original Sweave document.  @Bioconductor/sweave2rmd, do you think this is how to approach this or should we keep the output generated from the knit?

Reply:
The output of `sessionInfo()` should always be the output when the document was knit, i.e., an evaluated code chunk. I (not the author of the document, but primary maintainer of the package) do not think there is a need to format this as `toLatex(sessionInfo())`; just the output of `sessionInfo()`, generated dynamically, would be fine.